store/src/java/com/zimbra/cs/service/account/Auth.java

```diff
@@ -570,6 +573,15 @@ private Element doResponse(Element request, AuthToken at, ZimbraSoapContext zsc,
         if (acct.isIsMobileGatewayProxyAccount()) {
             response.addAttribute(AccountConstants.A_ZMG_PROXY, true);
         }
+
+        ToXML.encodeAttr(response, AccountConstants.E_TWO_FACTOR_AUTH_METHOD_ALLOWED, acct.getTwoFactorAuthMethodAllowed());
+        ToXML.encodeAttr(response, AccountConstants.E_TWO_FACTOR_AUTH_METHOD_ENABLED, acct.getTwoFactorAuthMethodEnabled());
+        ToXML.encodeAttr(response, AccountConstants.E_PREF_PRIMARY_TWO_FACTOR_AUTH_METHOD, acct.getPrefPrimaryTwoFactorAuthMethod());
+        String recoveryAddrs = acct.getPrefPasswordRecoveryAddress();
+        if (!StringUtil.isNullOrEmpty(recoveryAddrs)) {
+            String prefPwdRecoveryAddrs = Stream.of(recoveryAddrs.split(",")).map(add -> StringUtil.maskEmail(add)).collect(Collectors.joining(","));
+            ToXML.encodeAttr(response, AccountConstants.E_PREF_PASSWORD_RECOVERY_ADDRESS, prefPwdRecoveryAddrs);
```

Comment:
Please return masked zimbraPrefPasswordRecoveryAddress only when zimbraTwoFactorAuthMethodAllowed contains `email`, zimbraTwoFactorAuthMethodEnabled contains `email`, **and** zimbraPrefPasswordRecoveryAddressStatus is `verified`.
An email address set in zimbraPrefPasswordRecoveryAddress cannot be used for 2FA if the conditions are not met.
I'm sorry I missed checking the logic.

Reference: description in the ticket
`It also needs to contain a masked zimbraPrefPasswordRecoveryAddress if email method is allowed and enabled. (zimbraTwoFactorAuthMethodAllowed, zimbraTwoFactorAuthMethodEnabled, zimbraPrefPasswordRecoveryAddressStatus)`